source/deprecations/ember-data/v2.x.html.md

```diff
@@ -517,3 +517,95 @@ let author = this.store.peekRecord('user', 456);
 
 post.set('author', author);
 ```
+
+#### Arguments in Component Lifecycle Hooks
+
+##### until: 2.13.0
+##### id: ember-views.lifecycle-hook-arguments
+
+It is possible for component lifecycle hooks `didInitAttrs`, `didReceiveAttrs`, and `didUpdateAttrs` to receive arguments, but this functionality is part of private API, which is subject to change without notice. Those arguments will be removed because their use is harmful to component performance. However, since the arguments were accessible and some apps may be unknowingly depending on private API, this change will follow the formal deprecation process. Alternative approaches for all three hooks are below:
```

Comment:
Since this documentation will be post-deprecation, I think we should slightly re-word this. Something in the spirit of http://emberjs.com/deprecations/v2.x/#toc_initializer-arity, maybe.